kie-maven-plugin/src/main/java/org/kie/maven/plugin/BuildMojo.java

```diff
@@ -158,19 +208,53 @@ private void buildDrl() throws MojoFailureException {
 
         try {
             setSystemProperties(properties);
-            KieRepository kr = ks.getRepository();
-            InternalKieModule kModule = (InternalKieModule) kr.addKieModule(ks.getResources().newFileSystemResource(sourceFolder));
-            for (InternalKieModule kmoduleDep : kmoduleDeps) {
-                kModule.addKieDependency(kmoduleDep);
-            }
+            List<File> resources = getResourceFiles(sourceFolder);
+            ResultsImpl messages = null;
+            InternalKieModule kModule = null;
+            KieProject kieProject = null;
+            if (containsPMML) {
```

Comment:
@lanceleverich can you please clarify why we need 2 different types of build whether the kjar contains pmml or not? Can't we unify this?

Reply:
The primary reason that it done this way is that the BuildMojo did not make use of the KieBuilder, when compiling assets. The code that generates and compiles the Java classes, for PMML assets, is contained in the KieBuilderImpl. As we have talked about, that will change when we refactor the PMML into a service, similar to the approach taken by the DMN modules.